### DISCUSSION TITLE
[Bug]:  推送到仓库  vercel显示错误 不存在公共目录  求助解决方案

### DISCUSSION THREAD

**Discussion by smallSpoon** [CaiJimmy/hugo-theme-stack#666] [Q&A]
### What happened?

![部署出错](https://user-images.githubusercontent.com/105139527/184478379-95936741-ef04-4c5d-89c4-8df2928666ea.png)
有没有大佬遇到同类错误？上图

本地预览是没有问题的 下图
![image](https://user-images.githubusercontent.com/105139527/184479083-096a826d-1aef-42f1-8823-fb4acb326c7d.png)
。

跪谢！感谢了！试过很多回 也没有搜到类似的案例  小白实在不会 上来求助。

### Hugo version

0.89.0

### Theme version

最新

### Link to the page with bug

https://demo.stack.jimmycai.com

**Comment by zhixuan2333** [comment]
麻烦公开一下源仓库吗？没有源代码无法判断问题所在。

**Comment by smallSpoon** [comment]
> 麻烦公开一下源仓库吗？没有源代码无法判断问题所在。

私人仓库来的，是先在vercel上搭建成功后再自动链接github 生成的私人仓库

等我另外上传到新仓库。

**Comment by smallSpoon** [comment]
> 麻烦公开一下源仓库吗？没有源代码无法判断问题所在。

https://aying-five.vercel.app/OneDrive/      上传啦 麻烦了。

**Comment by zhixuan2333** [comment]
> <img width="631" alt="image" src="https://user-images.githubusercontent.com/59254886/185292427-910a1d81-6e9d-456f-abb2-e507480d06df.png">
`Framework` 要改到Hugo，他才会使用 hugo 的命令

感觉你的 screenshort 好怪啊，明明在用npm，他也安装Hugo，你确定和上传的URL的文件和你的仓储的文件一样吗？

## hugo-theme-stack-starter
如果不觉得麻烦的话，你可以尝试一下这个使用 @CaiJimmy 维护的 template。在最下边有写着关于vercel的设定。
https://github.com/CaiJimmy/hugo-theme-stack-starter

**Comment by smallSpoon** [comment]
> > <img alt="图像" width="631" src="https://user-images.githubusercontent.com/59254886/185292427-910a1d81-6e9d-456f-abb2-e507480d06df.png">
> 
> `Framework`要改到Hugo，他才会使用雨果 的命令
> 
> 感觉你的 screenshort 好怪啊，明明在用npm，他也安装Hugo，你确定和上传的URL的文件和你的仓储的文件一样吗？
> 
> ## 雨果-主题-堆栈-启动器
> 如果不觉得麻烦的话，你可以尝试一下这个使用 @CaiJimmy 维护的模板。在最下边有写着关于vercel的设定。 https://github.com/CaiJimmy/hugo-theme-stack-starter

我就是hugo 框架哦。 
![image](https://user-images.githubusercontent.com/105139527/186826376-644fa860-e25b-4dcc-bb9d-5f3eb24176ef.png)

**Comment by zhixuan2333** [comment]
你把Build command 的 override 关掉就行了
还有设定环境变量改 hugo 的版本。

**Comment by smallSpoon** [comment]
> 你把Build命令 的 override （重写） 关掉就行了 还有设定环境变量改 hugo 的版本。

![image](https://user-images.githubusercontent.com/105139527/187082604-c798a591-9314-41c1-ae65-0dd946064c4b.png)
全部关掉之后 就出现上述截图问题。所以之前才更改了Build命令 。

**Reply by zhixuan2333** [reply]
> Error: module "hugo-theme-stack" not found; either add it as a Hugo Module or store it in "/vercel/path0/themes".: module
does not exist

是因为主题的文件夹没有上传到GitHub上。

在blog的根目录运行这个命令 

```sh
git rm -f ./themes/hugo-theme-stack
git rm --cached ./themes/hugo-theme-stack
rm -rf ./themes/hugo-theme-stack
git submodule add https://github.com/CaiJimmy/hugo-theme-stack/ themes/hugo-theme-stack --force
git add .
git commit -am "add submodule hugo-theme-stack"
git push origin <master or main> # 你在 GitHub 上的 branch 的名字，一般是 master 或者是main
```

**Comment by CaiJimmy** [comment]
请把 https://github.com/smallSpoon/stack 改为公开仓库，方便排查问题。
